### DISCUSSION TITLE
Performance Comparison with heapq

### DISCUSSION THREAD

**Discussion by ShivKJ** [nvictus/priority-queue-dictionary#33] [Q&A]
@nvictus Do you have a runtime comparison of this library with `heapq`? Unlike `heapq`, which does not support changing the priority of items already in the priority queue, while `pqdict` does. Given that heapq only supports `push`, `pop`, `pushpop`, and poppush (`heapreplace`) operations, how does `pqdict` compare in terms of performance on these operations?

**Comment by nvictus** [comment]
The implementations of the core operators are based on the pure Python implementation of the [heapq module](https://github.com/python/cpython/blob/main/Lib/heapq.py) (which is the fallback to the C implementation) and should have similar performance on common operations with some overhead of keeping the position dictionary updated.

I've never done a formal benchmark but would welcome a PR with a benchmarking script.
